### ISSUE TITLE
Stack operation results summary contains config where secret variables are not hidden

### ISSUE THREAD

**Issue by Dysproz** [pulumi/pulumi#9776]
### What happened?

We've been using summary from stack operation result to print out summary of code execution with automation API.
However, it seems that summary (e.g. this one: https://github.com/pulumi/pulumi/blob/master/sdk/nodejs/automation/stack.ts#L710) contains config which contains config variables and even if variable is set to secret then it's not encrypted/hidden and easily accessible.

### Steps to reproduce

Run stack operation with automation API and gather operation result.

### Expected Behavior

Secret variables are not accessible in summary

### Actual Behavior

All secret variables vales are plain text.

### Versions used

```
Backend
Name           pulumi.com
URL            https://app.pulumi.com/Szymon
User           Szymon
Organizations  Szymon, starburstdata

NAME                                  VERSION
@pulumi/pulumi                        3.33.1
@pulumi/tls                           4.4.0
@types/node                           17.0.18
prettier                              2.5.1
typescript                            4.5.5
@pulumi/command                       0.1.0
yargs                                 17.4.1
@pulumi/aws                           5.4.0
@pulumi/kubernetes                    3.19.0
@types/jest                           27.5.1
@types/yargs                          17.0.10
jest                                  28.1.0
ts-jest                               28.0.3
@pulumi/eks                           0.37.1

```

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by jaxxstorm** [comment]
Chatting with @Frassle - this occurs because `UpdateSummary` is set to show all secrets by default:

https://github.com/pulumi/pulumi/blob/master/sdk/nodejs/automation/stack.ts#L555

This arguably should not be the default. Adding a flag to `UpdateSummary` https://github.com/pulumi/pulumi/blob/master/sdk/nodejs/automation/stack.ts#L646 is probably the best short term fix

**Comment by mikhailshilkov** [comment]
> Adding a flag to UpdateSummary https://github.com/pulumi/pulumi/blob/master/sdk/nodejs/automation/stack.ts#L646 is probably the best short term fix

We will take this short-term fix as it's non-breaking and will open a 4.0 issue to track a better way to handle outputs from Automation API.
